scrapy/extensions/feedexport.py

```diff
@@ -156,19 +156,29 @@ def _store_in_thread(self, file):
 
 class FTPFeedStorage(BlockingFeedStorage):
 
-    def __init__(self, uri):
+    def __init__(self, uri, active=False):
         u = urlparse(uri)
         self.host = u.hostname
         self.port = int(u.port or '21')
         self.username = u.username
         self.password = u.password
         self.path = u.path
+        self.active = active
+
+    @classmethod
+    def from_crawler(cls, crawler, uri):
+        return cls(
+            uri=uri,
+            active=crawler.settings['FEED_STORAGE_FTP_ACTIVE'] or False
+        )
 
     def _store_in_thread(self, file):
         file.seek(0)
         ftp = FTP()
         ftp.connect(self.host, self.port)
         ftp.login(self.username, self.password)
+        if self.active:
+            ftp.set_pasv(False)
```

Comment:
Any chances of adding a test case to cover this line?

Reply:
Sure @dangra but I am not sure what should I use in [FEEDTEST_FTP_URI and FEEDTEST_FTP_PATH](https://github.com/scrapy/scrapy/blob/4196d4869b8c5c14c9db19290d875fdd9958265d/tests/test_feedexport.py#L74). Should I add the unit test, start two FTP servers on my machine and test both passive and active method? Or Scrapy has a server to test it?